lib/index.ts

```diff
@@ -2,12 +2,26 @@ import * as path from 'path';
 import * as subProcess from './sub-process';
 import * as fs from 'fs';
 import * as tmp from 'tmp';
+import { SingleSubprojectInspectOptions } from '@snyk/cli-interface/dist/legacy/plugin';
+import { DepTree } from '@snyk/cli-interface/dist/legacy/common';
```

Comment:
this import looks weird? Why from dist?

Reply:
We do the same in Gradle plugin.
If a package does not export the types at the top level, they are only accessible via `dist/.../*.d.ts`.
Maybe there's a nicer way to reexport them at the top level (namespaces or something?), but I haven't found it yet.

Reply:
There must be something better than this, checking.

Reply:
we shouldn't do this, better to explicitly export things that are part of the public API (like these types) at the top level.

Reply:
https://github.com/snyk/snyk-cli-interface/pull/12/files >> PR with a fix

Reply:
Please also update here: https://github.com/snyk/snyk-gradle-plugin/pull/82 where the `dist` sneaked in

Reply:
First, I don't see a problem here. The import works, we clearly know how it works, it's clickable in the IDE, it's possible to discover the original definitions.

Second, there's no easy way around this.
Alternatives are:
* export types at the top level, so they lose qualifiers like `legacy/plugin` or `legacy/common` in the import path
* `import {legacy} from ....`, but then you would need to refer to every type as `legacy.plugin.Foo`. Too much repetition.
* aliases at the consumer side: `type Foo = legacy.plugin.Foo`. Again, repetition.
* tsconfig paths as per https://stackoverflow.com/a/38677886 - but I don't like shifting code into configuration files (and that means repetition at every consumer)
* modify the layout of `cli-interface` repo: get rid of `lib` and `dist`, drop everything at the top-level (and maybe even distribute is as `.ts` files)

Reply:
The first solution seems most sensible, put the qualifier in the name and then publish https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

So consumers just use the ambient `legacyDepTree` or similar. Regardless, this would need a new version (making the fix external to this PR), and I think we're really bikeshedding over this PR by dwelling on an import path folks.